### PR TITLE
Add conditionnals to detect gpsd API version

### PR DESCRIPTION
**Pull request by mobarre**
Attempted quick fix for issue #187. This will certainly require some testing (don't have a laptop GPS yet, so can't really test this).

### CODE REVIEW

**Comment by jochym** [comment]
Why we cannot make just the definition conditional? In the second change we read the same attribute and compare it with the same constant. I am sure you did it for some reason, I just would like to know that reason. ;)

**Comment by mobarre** [comment]
No, I have not done this for a reason, and the suggestions are very welcome ! I always make a point of matching any code style policy or coding preferences when contributing, so no problem.

My initial patch was worked out in a few minutes... It probably needs some refining.

**Comment by mobarre** [comment]
No, I have not done this for a reason, and the suggestions are very welcome ! My initial patch was worked out in a few minutes... It probably needs some refining.
Would this do: a single test in gps_driver.h + a define and ifdefs in the cpp file ? or even better a test through cmake and only ifdefs in gps_driver.cpp ?

File: indi-gpsd/gps_driver.cpp [hunk 1]
```diff
@@ -279,8 +282,11 @@ IPState GPSD::updateGPS()
             break;
     }
 
-
+#if GPSD_API_MAJOR_VERSION >= 10
+    if (gpsFix->status == STATUS_NO_FIX)
```
Comment (jpaana) [inline indi-gpsd/gps_driver.cpp:286]: This change is bogus as the pointer is not initialized anywhere. The status field has indeed moved so the check should be:
if (gpsData->fix.status == STATUS_NO_FIX)

**Pull request merged**